app.py

```diff
@@ -1,26 +1,402 @@
-from modules.Preprocessing import get_tweet_data
+from pprint import pprint
+from operator import itemgetter
+from typing import List
+from datetime import datetime, timedelta
+from modules.web_scraper import get_tweets
+from modules.Preprocessing import handle_tweet_data
 from modules.Sentiment_Analysis import train_model_if_necessary, analyze_many_tweets
+from modules.presentation import get_tweets_in_daterange, get_by_key_value, remove_sentiment, get_sentiment, bar_plot, line_plot, pie_chart
+import argparse
+import re
 
-# Train model
-train_model_if_necessary()
-# scrape here
-print('Scraping...')
-tweet_data = [{}]  # do stuff
-# open file for preprocessing
-print('Preprocessing...')
-# FILENAME = './'
-# with open(FILENAME) as file_object:
-#     tweet_data = file_object.read()
+_REGEX_CHAR_MATCHER_HASHTAGS = re.compile('[^A-Za-z0-9]')
 
-handled_tweets = get_tweet_data(tweet_data)  # do stuff
+#########CUSTOM TYPES#########
 
-# analysis here
-print('Performing sentiment analysis...')
-analyzed_tweets = *handled_tweets  # do stuff
 
-# presentation here
-print('Generating presentations...')
-result = *analyzed_tweets  # do stuff
+def _restricted_float(val: float):
+    """
+    Only allow float values within our range [0.00-1.00]
+    """
+    try:
+        val = float(val)
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"{val} not a floating-point literal")
```

Comment:
I still don't understand this. Is it needed when deciding the high and low % of sentiment?

Reply:
Yes, it is related to the `uncertainity_low` and `uncertainty_high` that can be passed to Sentiment Analysis.

Arguments can have types passed to them, specifically for this example (as seen in line 378, 385):

```py
parser.add_argument(
        '-cl',
        help="The lower float value for determining if a sentiment is deemed uncertain.\nEXAMPLE: 0.15.\nVALUES: [0.00-1.00]",
        type=_restricted_float,
        ...
```
What we could do here, is to pass `type=float` and argparse would try to parse the argument to a float. If it could not, it would throw an error.  
That is all good, and essentially what we are doing here - replicating the behavior of the argparse parser. The important part of the function is below the highlighted part, where we check if the float value is within the allowed range:

```py
if 0.0 < val > 1.0:
        raise argparse.ArgumentTypeError(f"{val} not in range [0.0, 1.0]")
    return val
```